Src/HALAL/Models/Runes/Runes.cpp

```diff
@@ -0,0 +1,90 @@
+map<uint8_t,Pin> DigitalInput::service_ids = {};
+
+// ---------- DigitalOutput ----------
+forward_list<uint8_t> DigitalOutput::id_manager = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255};
+map<uint8_t,Pin> DigitalOutput::service_ids = {};
+#endif
+
+#ifdef HAL_TIM_MODULE_ENABLED
+// ---------- Encoder ----------
+extern TIM_HandleTypeDef htim8;
+
+forward_list<uint8_t> Encoder::id_manager = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255};
```

Comment:
No me gusta como se ve este archivo.

- id manager por ejemplo deberia estar dentro de cada archivo correspondiente no aqui. 
- ¿Que sentido tiene poner los extern aqui?
- Deberias dejar aqui solo los mapas que instancias.
- Los `#ifdef` por debajo del comentario que indica que es.
- El comentario de que es cada cosa.

Los comentarios asi:
```suggestion
/************************************************
 *                     PWM
 ***********************************************/
```

Reply:
1. Es bastante subjetivo, al final es una cola que es igual en todos y de esta forma solo se quedan las funciones en el service.cpp
2. Si no los pones no compila, en los mapas se usan los htim.
3. Lo mismo que el 1, pero si es subjetivo lo hablamos hoy no me importa cambiarlo
4. Vale si, el ifdef lo cambio
5. No entiendo exactamente a que te refieres con poner un comentario de que es cada cosa

Reply:
1. Dejalo en los archivos que les toca por eso mismo, es algo especifico de cada uno no como las otras estructuras que si las pueden usar otras cosas.
2. Tienes razon.
3. Deja solo lo que pueda ser general.
4. Fino.
5. Lo que tienes `//-----------------------PWM--------------------------` lo pongas como te he puesto.